# Урок 2

In [95]:
from math import factorial, exp

def combinations(k, n):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

def bernulli(k, n, p):
    return combinations(k,n) * p**k * (1-p)**(n-k)

def puasson_formula(n, p, m):
    lambda_p = n*p
    puasson = (lambda_p**m/factorial(m)) * exp(-lambda_p)
    return puasson

## №1
Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Формула Бернулли $P_n(X=k) = C_n^k p^k q^{n-k}$

n = 100

k = 85

p = 0.8

$P_{k=85} = C_{100}^{85} \cdot 0.8^{85} \cdot 0.2^{15} \approx 0.048 \approx 4.8$%

In [96]:
bernulli(85, 100, 0.8)

0.048061793700746355

## №2

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. 

Какова вероятность, что ни одна из них не перегорит в первый день?

По формуле Пуассона $P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$

$n = 5000$

$m = 0$

$p = 0.0004$

$λ = np = 5000*0.0004 = 2$

$P_0 \approx \frac{2^0}{0!}2.71^{-2} \approx 13.6$%

In [97]:
puasson_formula(5000, 0.0004, 0)

0.1353352832366127

По формуле Бернулли $P_n(X=k) = C_n^k p^k q^{n-k}$

$k = 0$

$n = 5000$

$p = 0.0004$

$P_{k=0} = C_{5000}^{0} \cdot 0.0004^{0} \cdot (1-0.0004)^{5000} \approx 0.135 \approx 13.5$%

In [98]:
bernulli(0, 5000, 0.0004)

0.13528114551440706

б) Какова вероятность, что перегорят ровно две?

По формуле Пуассона $P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$

$n = 5000$

$m = 2$

$p = 0.0004$

$λ = np = 5000*0.0004 = 2$

$P_2 \approx \frac{2^2}{2!}2.71^{-2} \approx 27.2$%

In [99]:
puasson_formula(5000, 0.0004, 2)

0.2706705664732254

По формуле Бернулли $P_n(X=k) = C_n^k p^k q^{n-k}$

$k = 2$

$n = 5000$

$p = 0.0004$ 

$P_{k=2} = C_{5000}^{2} \cdot 0.0004^{2} \cdot (1-0.0004)^{4998} \approx 0.27 \approx 27$%

In [100]:
bernulli(2,5000,0.0004)

0.2707247150266753

## №3
Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Формула Бернулли $P_n(X=k) = C_n^k p^k q^{n-k}$

$k = 70$

$n = 144$

$p = 0.5$

$P_{k=70} = C^{70}_{144} \cdot 0.5^{70} \cdot 0.5^{74} \approx 0.0628 \approx 6.28$%

In [101]:
bernulli(70, 144, 0.5)

0.06281178035144776

## №4
В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.

Какова вероятность того, что все мячи белые?

Вероятность того, что 2 вытащенных мяча из первого ящика будут белыми:

$C^{2}_{7} = \frac{7!}{2!(7 - 2)!} = 21$

$C^{2}_{10} = \frac{10!}{2!(10 - 2)!} = 45$

$\frac{21}{45} \approx 0.47 \approx 47\% $

In [102]:
x1 = combinations(2, 7) / combinations(2, 10)
x1

0.4666666666666667

Вероятность того, что 2 вытащенных мяча из второго ящика будут белыми:

$C^{2}_{9} = \frac{9!}{2!(9 - 2)!} = 36$

$C^{2}_{11} = \frac{11!}{2!(11 - 2)!} = 55$

$\frac{36}{55} \approx 0.65 \approx 65\% $

In [103]:
x2 = combinations(2, 9) / combinations(2, 11)
x2

0.6545454545454545

Вероятность того что все мячи будут белые 

$0.47*0.65 = 0.3055 = 30.5\%$

In [104]:
X = x1 * x2
X

0.3054545454545455

Какова вероятность того, что ровно два мяча белые?

Два белых мяча из первого ящика

In [105]:
y1 = (combinations(2,7) / combinations(2,10)) * (combinations(2,2) / combinations(2,11))
y1

0.008484848484848484

Вероятность того, что оба белых мяча из первого ящика = **0.8%**

Два белых мяча из второго ящика

In [106]:
y2 = (combinations(2,3) / combinations(2,10)) * (combinations(2,9) / combinations(2,11))
y2

0.04363636363636363

Вероятность того, что оба белых мяча из второго ящика = **4.3%**

Один белый мяч из первого ящика и один белый мяч из второго ящика

In [107]:
y3 = (combinations(1,7)*combinations(1,3) / combinations(2,10)) * (combinations(1,9)*combinations(1,2) / combinations(2,11))
y3

0.15272727272727274

Вероятность того, что один белый мяч из первого ящика и один белый мяч из второго ящика = **15.3%**

In [108]:
Y = y1 + y2 + y3
Y

0.20484848484848486

вероятность того, что ровно два мяча белые = **20.4%**

Какова вероятность того, что хотя бы один мяч белый?

In [109]:
z1 = (combinations(2,3) / combinations(2,10)) * (combinations(2,2) / combinations(2,11))
1 - z1

0.9987878787878788

Вероятность того, что хотя бы один мяч белый =  **99.8%**